### PR TITLE
Use NSURLSessionDataTask to download remote data

### PR DESCRIPTION
**Pull request by softwarebouwer**
I also had issue #154, it seemed the NSStream implementation only worked for plain http requests. My ObjC is somewhat rusty, but I removed NSStream and implemented NSURLSessionDataTask.

### CODE REVIEW

**Comment by softwarebouwer** [comment]
Noticed a small error, I checked the wrong error line 84 should be:
`if (taskError || !data || data.length == 0) {`

**Comment by bisherbas** [comment]
Any idea when will be this PR merged, if at all?

**Comment by p1g30n** [comment]
@2ndalpha is there a way to build it on 10.10? If yes I can test. 
Otherwise any chance for a test release?

**Comment by kintaro1981** [comment]
as @2ndalpha don't seams to have time for this? @softwarebouwer can you build a release with your PR included?

**Comment by softwarebouwer** [comment]
> as @2ndalpha don't seams to have time for this, please @softwarebouwer can you build a release with your PR included?

This is the build I've been using since I created the PR
[Gas Mask-0.8.6-PR155.zip](https://github.com/2ndalpha/gasmask/files/2792102/Gas.Mask-0.8.6-PR155.zip)

**Comment by p1g30n** [comment]
@softwarebouwer thanks for posting this. Works well on 10.10.5, including the added SSL support. 
+1 vote for a release.

**Comment by MadMacMad** [comment]
@softwarebouwer 
many thx for the upload... but it crashes all time on Mojave ...
any chance for a updated version?
many thx 
respect
M.

**Comment by softwarebouwer** [comment]
> @softwarebouwer
> many thx for the upload... but it crashes all time on Mojave ...
Strange it was crashing for you, because I’ve only been using it on Mojave.

**Comment by MadMacMad** [comment]
> @softwarebouwer
> many thx for the upload... but it crashes all time on Mojave ...
> Strange it was crashing for you, because I’ve only been using it on Mojave.

i found the problem...
after doing this all works fine...
sudo /bin/chmod +a 'user:YOUR_USERNAME:allow write' /private/etc/hosts

**Comment by softwarebouwer** [comment]
> can you make build on Mojave so that we get the dark theme too?
Maybe later, I'm on vacation at the moment :-)

File: Source/HostsDownloader.m [hunk 1]
```diff
@@ -25,8 +25,8 @@
 #import "ExtendedNSApplication.h"
 
 #define kBufferSize 1024
```
[the remark this answers is not in the document]
Comment (softwarebouwer) [reply Source/HostsDownloader.m]: Done, and I also removed #import "ExtendedNSString.h".

**Pull request merged**